pkg/provisioner/ironic/ironic.go

```diff
@@ -354,6 +354,39 @@ func (p *ironicProvisioner) findExistingHost() (ironicNode *nodes.Node, err erro
 
 }
 
+func (p *ironicProvisioner) updateAutomatedClean(ironicNode *nodes.Node) (err error) {
+	p.log.Info("setting cleaning mode",
+		"ID", ironicNode.UUID,
+		"mode", p.host.Spec.AutomatedCleaningMode)
+
+	p.publisher("CleaningMode", string(p.host.Spec.AutomatedCleaningMode))
+
+	value := p.host.Spec.AutomatedCleaningMode != metal3v1alpha1.CleaningModeDisabled
+
+	updates := nodes.UpdateOpts{
+		nodes.UpdateOperation{
+			Op:    nodes.ReplaceOp,
+			Path:  "/automated_clean",
+			Value: value,
+		},
+	}
+
+	_, err = nodes.Update(p.client, ironicNode.UUID, updates).Extract()
+
+	switch err.(type) {
+	case nil:
+	case gophercloud.ErrDefault409:
+		p.log.Info("could not update ironic node automated_clean, busy")
+		_, err = retryAfterDelay(provisionRequeueDelay)
```

Comment:
This line just returns nil and does not retry. This is wrong and will cause the calling code to believe that cleaning has been disabled when it hasn't.

Reply:
@zaneb you are right. Do you think I should not retry at all, and replace L380 with for example something like
`return errors.Wrap(err, "could not update ironic node automated_clean, busy")` or shall I instead add `result provisioner.Result` as second return value for `updateAutomatedClean()` and do retry? Maybe the latter option makes more sense, because I can see that we are retrying in most of the places in ironic.go.

Reply:
Returning an `error` will result in an immediate* requeue plus an ugly error in the logs. Generally speaking, responding to ironic getting hammered by hammering it harder is not ideal, which is why we use retryAfterDelay everywhere else we get a 409.

\* subject to throttling in the controller

Reply:
Fixed accordingly, thank you!